modules/pricer/src/main/java/com/opengamma/strata/pricer/cms/SabrExtrapolationReplicationCmsPeriodPricer.java

```diff
@@ -266,11 +271,10 @@ public double adjustmentToForwardRate(
       RatesProvider provider,
       SabrParametersSwaptionVolatilities swaptionVolatilities) {
 
-    ArgChecker.isTrue(
-        cmsPeriod.getFixingDate().isAfter(provider.getValuationDate()),
-        "Adjustment computed only for coupon with fixing (strictly) after the valuation date");
-    double adjustedForwardRate = adjustedForwardRate(cmsPeriod, provider, swaptionVolatilities);
-    double forward = swapPricer.parRate(cmsPeriod.getUnderlyingSwap(), provider);
+    // Create the coupon equivalent to the cap or floor.
+    CmsPeriod coupon = cmsPeriod.toBuilder().floorlet(null).caplet(null).build();
```

Comment:
Suggest adding a method to `CmsPeriod` to do this, to hide the use of nulls. Maybe `toCouponEquivalent()`?